Add Z-Wave overview and setup sections

c0ffeeca7 opened the pull request:
- add overview section
- expand setup procedure

## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

c0ffeeca7 committed:
Add overview and setup sections

- add overview section explaining the terminology around Z-Wave
- expand setup procedure

c0ffeeca7 committed:
Fix typo in Z-Wave

c0ffeeca7 committed:
Fix more typos

MartinHjelmare commented on source/_integrations/zwave_js.markdown:
```diff
@@ -44,18 +44,48 @@ ha_integration_type: hub
+## Overview
 
-{% include integrations/config_flow.md %}
+The Z-Wave integration in Home Assistant talks to one of two possible add-ons: the official Z-Wave JS add-on, or the community add-on Z-Wave JS UI. Both add-ons include the Node Z-Wave JS driver. This driver talks to a controller device (Z-Wave dongle or GPIO (general purpose input/output) board)
```
It reads a bit weird that we say here that the integration talks to one of two add-ons, and then further below under the advanced instructions we say that you can also do without any add-on.

I think we should not talk about add-ons at all in the overview. I think we should just list the three requirements, a controller, the Z-Wave JS server and the Z-Wave integration.

Then in the next section we explain that the easiest way to get started running the server is to use the official add-on. The other ways of running the server are explained in the advanced installation instructions.

c0ffeeca7 replied:
Thank you for the input, @MartinHjelmare! I felt the need to introduce some of the terminology around Z-Wave, as I was confused about it during the installation procedure. But I realize that it is a lot of implementation detail to start with. I removed the section.

AlCalzone commented on source/_integrations/zwave_js.markdown:
```diff
@@ -44,18 +44,48 @@ ha_integration_type: hub
 ---
 
-This integration allows you to control a Z-Wave network via the [Z-Wave JS](https://zwave-js.github.io/node-zwave-js/#/) driver.
+The Z-Wave integration allows you to control a Z-Wave network via the [Node Z-Wave JS](https://zwave-js.github.io/node-zwave-js/#/) driver.
```
```suggestion
The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS](https://zwave-js.github.io/node-zwave-js/#/) driver.
```
There's no "Node" in the official name, just in the URL (because it is a Node.js project)

AlCalzone commented on source/_integrations/zwave_js.markdown:
```diff
@@ -44,18 +44,48 @@ ha_integration_type: hub
+
+The easiest way to get started is by using the built-in Z-Wave JS add-on in Home Assistant.
+
+If you run Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Z-Wave JS Server add-on, refer to the [advanced installation instructions](#advanced-installation-instructions).
```
I don't think the addon is called "Z-Wave JS Server add-on"

c0ffeeca7 committed:
Terminology: Fix error in driver name

Co-authored-by: AlCalzone <d.griesel@gmx.net>

frenck commented on source/_includes/integrations/manual_config_steps.md:
```diff
@@ -0,0 +1,31 @@
+{% capture name %}{{ include.name | default: page.title }}{% endcapture %}
```
Unless we have more cases, we should not extract things.

At this point, there is only a single use case for this one, so we should not make it a generic include?

c0ffeeca7 committed:
Implement review feedback

- by @MartinHjelmare, @AlCalzone, @Frenck
- Remove part of overview section
 (too much implementation detail at this stage)
- Fix terminology inconsistencies
- Remove unnecessary include

c0ffeeca7 committed:
Remove unnecessary introductory clause

c0ffeeca7 committed:
Merge branch 'current' into z-wave-update-setup-server-procedure

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -48,40 +48,57 @@ The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS
+{% details "Manual setup steps" %}
+Use this My button:
+
+{% my config_flow_start badge domain=domain %}, or follow these steps:
```
```suggestion
{% my config_flow_start badge domain="zwave_js" %}, or follow these steps:
```

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -48,40 +48,57 @@ The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS
+* Browse to your Home Assistant instance.
+* In the sidebar click on _**{% my config icon %}**_.
+* From the configuration menu select: _**{% my integrations %}**_.
+{% if page.ha_integration_type == 'helper' %}
+* In top of the screen click the tab: _**{% my helpers %}**_.
+* In the bottom right, click on the
+  _**{% my config_flow_start icon domain=domain title="Create helper" %}**_ button.
+{% else %}
```
```suggestion
```

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -48,40 +48,57 @@ The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS
+{% else %}
+* In the bottom right, click on the
+  _**{% my config_flow_start icon domain=domain %}**_ button.
+{% endif %}
```
```suggestion
```

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -48,40 +48,57 @@ The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS
+* In the bottom right, click on the
+  _**{% my config_flow_start icon domain=domain %}**_ button.
+{% endif %}
+* From the list, search and select _**"{{ name }}"**_.
```
```suggestion
* From the list, search and select _**"Z-Wave"**_.
```

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -48,40 +48,57 @@ The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS
+  _**{% my config_flow_start icon domain=domain title="Create helper" %}**_ button.
+{% else %}
+* In the bottom right, click on the
+  _**{% my config_flow_start icon domain=domain %}**_ button.
```
```suggestion
  _**{% my config_flow_start icon domain="zwave_js" %}**_ button.
```

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -48,40 +48,57 @@ The Z-Wave integration allows you to control a Z-Wave network via the [Z-Wave JS
-   * If your dongle is not recognized, follow these steps: {% include integrations/manual_config_steps.md %}
+   * If your dongle is not recognized, follow these steps:
+
+{% capture name %}{{ include.name | default: page.title }}{% endcapture %}
+{% capture domain %}{{ include.domain | default: page.ha_domain }}{% endcapture %}
+
```
```suggestion
```

c0ffeeca7 committed:
Update source/_integrations/zwave_js.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Remove redundant steps

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Remove redundant element

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Use specific name instead of generic variable

Variable is not needed here, as it is only used in this specific case.

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Use specific name instead of variable

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Remove redundant code

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -44,18 +44,59 @@ ha_integration_type: hub
+To run a Z-Wave network, you need the following elements:
 
-If you run Home Assistant Container, Home Assistant Core, or you don't want to use the built-in Z-Wave JS Server add-on, please see the [advanced installation instructions](#advanced-installation-instructions).
+* A [supported Z-Wave controller](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). First time user? For recommendations on what to buy, go [here](integrations/zwave_js/#which-z-wave-controller-should-i-buy).
```
```suggestion
* A [supported Z-Wave controller](/docs/z-wave/controllers/#supported-z-wave-usb-sticks--hardware-modules). First-time user? For recommendations on what to buy, go [here](integrations/zwave_js/#which-z-wave-controller-should-i-buy).
```

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -44,18 +44,56 @@ ha_integration_type: hub
+
+{% enddetails %}
+
+3. Wait for the installation to complete.
```
Something happend with the numbering here

c0ffeeca7 replied:
I was trying to use a workaround. If I put 1. here, the numbering starts over:

![image](https://user-images.githubusercontent.com/38767475/230018212-52648f53-8442-466a-a755-d4eac76e64ef.png)

it seems that %details% breaks up the list. I was not able to find a way around that.

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -44,18 +44,56 @@ ha_integration_type: hub
 
 <p class='note'>
-While your Z-Wave mesh is permanently stored on your stick, the additional metadata is not. When the Z-Wave integration starts up the first time, it will interview your entire Z-Wave network. Depending on the number of devices paired with the Z-Wave stick, this can take a while. You can speed up this process by manually waking up your battery-powered devices. Most of the time this is a press on t ...
+While your Z-Wave mesh is permanently stored on your dongle, the additional metadata is not. When the Z-Wave integration starts up the first time, it will interview your entire Z-Wave network. Depending on the number of devices paired with the Z-Wave dongle, this can take a while. You can speed up this process by manually waking up your battery-powered devices. Most of the time, this is a press o ...
```
```suggestion
While your Z-Wave mesh is permanently stored on your dongle, the additional metadata is not. When the Z-Wave integration starts up the first time, it will interview your entire Z-Wave network. Depending on the number of devices paired with the Z-Wave dongle, this can take a while. You can speed up this process by manually waking up your battery-powered devices. Most of the time, this is a button press on those devices (see their manual). It is not necessary to exclude and re-include devices from the mesh.
```

frenck commented on source/_integrations/zwave_js.markdown:
```diff
@@ -557,28 +597,28 @@ All of these features can be accessed either in the Z-Wave integration configura
+* **Add device:** Allows you to pre-provision a SmartStart device or start the inclusion process for adding a new device to your network.
+* **Remove device:** Starts the exclusion process for removing a device from your network.
+* **Heal network:** Forces your network to rediscover routes to the controller from each device. This is useful when devices or the controller have moved to a new location, or if you are having significant problems with your network, but it also generates a lot of network traffic and should be used sparingly.
+* **Third party data opt-in/out:** Allows you to opt-in or out of telemetry that the zwave-js project collects to help inform development decisions, influence manufacturers, etc. This telemetry is disabled by default and has to be opted in to be activated.
```
```suggestion
* **Third-party data opt-in/out:** Allows you to opt-in or out of telemetry that the Z-Wave JS project collects to help inform development decisions, influence manufacturers, etc. This telemetry is disabled by default and has to be opted in to be activated.
```

c0ffeeca7 committed:
Fix typo

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Rephrase

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

c0ffeeca7 committed:
Fix terminology

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

MartinHjelmare commented:
Nice! Text wise it's good for me now.

If we still want to show the figure with the topology of the different parts involved when using the integration we could perhaps show it under the advanced installation instructions?

c0ffeeca7 commented:
thank you for your help, @MartinHjelmare, @AlCalzone, @frenck :+1:
